odk2gn/gn2_utils.py

```diff
@@ -1,106 +1,131 @@
-                n_fields = n_fields + get_nomenclatures_fields(
-                    module_code=module.module_code,
-                    niveau=niveau
-                )
-
-            nomenclatures = get_nomenclature_data(n_fields)
-            files['gn_nomenclature.csv'] = to_csv(
-                header=("mnemonique", "id_nomenclature", "cd_nomenclature", "label_default"),
-                data=nomenclatures
+        files["gn_taxons.csv"] = to_csv(header=("cd_nom", "nom_complet", "nom_vern"), data=data)
+    # Observers
+    if not skip_observers:
+        data = get_observer_list(module.id_list_observer)
+        files["gn_observateurs.csv"] = to_csv(header=("id_role", "nom_complet"), data=data)
+    # JDD
+    if not skip_jdd:
+        data = format_jdd_list(module.datasets)
+
+        files["gn_jdds.csv"] = to_csv(header=("id_dataset", "dataset_name"), data=data)
+    # Sites
+    if not skip_sites:
+        data = get_site_list(module.id_module)
+        files["gn_sites.csv"] = to_csv(
+            header=("id_base_site", "base_site_name", "geometry"), data=data
+        )
+
+    if not skip_sites_groups:
+        data = get_site_groups_list(module.id_module)
+        files["gn_groupes.csv"] = to_csv(header=("id_sites_group", "sites_group_name"), data=data)
+
+    # Nomenclature
+    if not skip_nomenclatures:
+        n_fields = []
+        for niveau in ["site", "visit", "observation"]:
+            n_fields = n_fields + get_nomenclatures_fields(
+                module_code=module.module_code, niveau=niveau
             )
-        return files
+
+        nomenclatures = get_nomenclature_data(n_fields)
+        files["gn_nomenclatures.csv"] = to_csv(
+            header=("mnemonique", "id_nomenclature", "cd_nomenclature", "label_default"),
+            data=nomenclatures,
+        )
+    return files
+
+
+def get_site_groups_list(id_module: int):
+    """Return dict of TMonitoringSitesGroups
+
+    :param id_module: Identifier of the module
+    :type id_module : int"""
+
+    data = (
+        DB.session.query(TMonitoringSitesGroups)
+        .order_by(TMonitoringSitesGroups.sites_group_name)
+        .filter_by(id_module=id_module)
+        .all()
+    )
+
+    return [group.as_dict() for group in data]
 
 
 def get_taxon_list(id_liste: int):
-    """Return tuple of Taxref for id_liste
+    """Return dict of Taxref
 
     :param id_liste: Identifier of the taxref list
     :type id_liste: int
     """
     data = (
-        DB.session.query(Taxref.cd_nom, Taxref.nom_complet, Taxref.nom_vern)
+        DB.session.query(Taxref)
+        .order_by(Taxref.nom_complet)
         .filter(BibNoms.cd_nom == Taxref.cd_nom)
         .filter(BibNoms.id_nom == CorNomListe.id_nom)
         .filter(CorNomListe.id_liste == id_liste)
-        .all()
+        .limit(3000)
     )
-    return data
+    taxons = []
+    for tax in data:
+        tax = tax.as_dict()
+        tax["nom_complet"] = tax["nom_complet"] + " - " + tax["nom_vern"]
```

Comment:
Soulève une erreur `TypeError: can only concatenate str (not "NoneType") to str` lorsque nom_vern est null (fréquent dans TaxRef)